Integrations/integration-VirusTotal-Private_API.yml

```diff
@@ -551,98 +553,120 @@ script:
         """
 
         args = demisto.args()
-        url = args.get('resource')
+        urls = argToList(args.get('resource'))
         all_info = 1 if args.get('allInfo', None) == 'true' else 0
         threshold = int(args.get('threshold', None) or demisto.params().get('urlThreshold', None) or 10);
         short_format = args.get('shortFormat', None) == 'true'
         full_response = args.get('fullResponse', None) == 'true'
+        retries_left = int(args.get('retries', 2))
         if (full_response):
             max_len = 1000
         else:
             max_len = 50
 
-        response = get_url_report(url, all_info)
-
-        if (response.get('response_code', None) == -1):
-            return "Invalid url"
+        completed_responses = []
+        completed_responses_dict = []
+        non_completed_urls = []
+        for url in urls:
+            response = get_url_report(url, all_info)
 
-        positives = demisto.get(response, 'positives')
+            if (response.get('response_code', None) == -1):
+                return "Invalid url"
 
-        md = '## VirusTotal URL report for: ' + url + '\n'
-        md += 'Scan ID: **' + str(response.get('scan_id', '')) + '**\n'
-        md += 'Scan date: **' + str(response.get('scan_date', '')) + '**\n'
-        md += 'Detections / Total: **' + str(positives) + '/' + str(response.get('total', '')) + '**\n'
-        md += 'VT Link: [' + str(response.get('resource', '')) + '](' + str(response.get('permalink', '')) + ')\n'
-
-        ec = {}
-        ec['DBotScore'] = []
-        dbotScore = 0
+            if (not response.get('total')):
+                non_completed_urls.append(url)
+            else:
+                completed_responses_dict.append({
+                    'response': response,
+                    'url': url
+                })
+                completed_responses.append(response)
+
+        while non_completed_urls and retries_left:
```

Comment:
seems there is no place where `non_completed_urls` is being emptied, so if retry is greater then 1, it will happen anyway, if it is unneeded.

maybe it is simpler just to create a dict of key that is url and value that is a response, then you can iterate on every one of them and retry if there is no response yet.
this way you can hold all of the relevant data in one place.

Reply:
you're right, this is a relic from when it was an if and ran just once, I'll fix this